src/kernel/arch/x86_64/gdt.h

```diff
@@ -4,22 +4,28 @@
 #include <stdint.h>
 #include <macro.h>
 
-struct gdt_entry
+typedef struct packed
 {
-    u16int base_low;
-    u16int limit_low;
-    u8int access_byte;
-    u8int base_middle;
-    u8int base_high;
-}__attribute__((packed));
-typedef struct gdt_entry gdt_entry_t;
-
-struct gdt_ptr
+    uint16_t *limitLow;
+    uint16_t *baseLow;
+    uint8_t *baseMiddle;
+    uint8_t *baseHigh;
+    uint8_t *granularity;
+    uint8_t *access;
+    uint8_t *flags;
+
+}GDTEntry;
+
+typedef struct packed
 {
-    u16int limit;
-    uintptr_t base;
-}__attribute__((packed));
-typedef struct gdt_ptr gdt_ptr_t;
+    uintptr_t *limit;
+    uint16_t *base;
```

Comment:
why ptr ?

Reply:
I can change with other type if you want. With uint32_t.